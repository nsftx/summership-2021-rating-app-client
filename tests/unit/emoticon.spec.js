import { mount, createWrapper } from '@vue/test-utils'
import Emoticon from '../../src/component/emoticon'
describe('Emoticon', () => {
  const wrapper = mount(Emoticon, {
    propsData: {
      data: {
        src: 'positive-vote.svg',
        color: 'rgb(0, 168, 107)',
        id: 1
      }
    }
  })
  test('Component exist', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('should have vote method', function () {
    expect(typeof wrapper.vm.vote).toBe('function')
  })
  it('should emit userVoted events', function () {
    wrapper.vm.vote()
    expect(wrapper.emitted().userVoted).toBeTruthy()
  })
  it('should emit post event', function () {
    wrapper.vm.vote()
    const rootWrapper = createWrapper(wrapper.vm.$root)
    expect(rootWrapper.emitted().post).toBeTruthy()
  })
})
