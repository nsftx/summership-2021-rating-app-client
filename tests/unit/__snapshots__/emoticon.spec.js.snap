// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoticon renders correctly 1`] = `
<div
  class="main"
>
  <div
    class="left-border"
    style="background-color: rgb(0, 168, 107);"
  />
   
  <div
    class="container"
  >
    <div
      class="background"
      style="background-color: rgb(0, 168, 107);"
    />
     
    <img
      alt="Emoticon"
      src="positive-vote.svg"
    />
  </div>
</div>
`;
